tools/cldr-apps/src/org/unicode/cldr/web/MailSender.java

```diff
@@ -378,12 +380,11 @@ public void run() {
                 conn = db.getDBConnection();
                 conn.setAutoCommit(false);
                 java.sql.Timestamp sqlnow = DBUtils.sqlNow();
-                s = DBUtils.prepareForwardUpdateable(conn, "select * from " + CLDR_MAIL + " where sent_date is NULL and id > ?  and try_count < 3 order by id "
+                s = DBUtils.prepareForwardUpdateable(conn, "select * from " + CLDR_MAIL + " where sent_date is NULL and id > ? and try_count < 3 order by id "
                     + (DBUtils.db_Mysql ? "limit 1" : ""));
                 s.setInt(1, lastIdProcessed);
                 rs = s.executeQuery();
-
-                if (rs.first() == false) {
+                if (rs.next() == false) { // No mail to send.
```

Comment:
this is the core change.  A forward-only cursor can't go "back" to next, and it turns out it is formally an error, even if 'rs.next()' is the first call.

There may be more instances of this, but this is the only  one i found.